test/org/sagebionetworks/bridge/services/AccountWorkflowServiceTest.java

```diff
@@ -228,23 +318,47 @@ public void requestResetPasswordInvalidEmailFailsQuietly() throws Exception {
     }
 
     @Test
+    public void requestRestPasswordUnverifiedEmailFailsQuietly() throws Exception {
+        when(service.createTimeLimitedToken()).thenReturn("ABC");
+        when(mockAccount.getEmail()).thenReturn(EMAIL);
+        when(mockAccountDao.getAccount(ACCOUNT_ID_WITH_EMAIL)).thenReturn(mockAccount);
+        
+        service.requestResetPassword(study, ACCOUNT_ID_WITH_EMAIL);
+        
+        verifyNoMoreInteractions(mockSendMailService);
+        verifyNoMoreInteractions(mockNotificationsService);
+    }
+    
+    @Test
+    public void requestRestPasswordUnverifiedPhoneFailsQuietly() throws Exception {
+        when(service.createTimeLimitedToken()).thenReturn("ABC");
+        when(mockAccount.getPhone()).thenReturn(TestConstants.PHONE);
+        when(mockAccountDao.getAccount(ACCOUNT_ID_WITH_PHONE)).thenReturn(mockAccount);
+        
+        service.requestResetPassword(study, ACCOUNT_ID_WITH_PHONE);
+        
+        verifyNoMoreInteractions(mockSendMailService);
+        verifyNoMoreInteractions(mockNotificationsService);
+    }
+    
+    @Test
     public void resetPassword() {
-        when(mockCacheProvider.getString("sptoken:api")).thenReturn(EMAIL);
+        when(mockCacheProvider.getString("sptoken:email:api")).thenReturn(EMAIL);
```

Comment:
Unit test for resetting password via phone? (Including invalidating the phone cache key.)

Reply:
Added